transport/src/main/java/io/netty/channel/socket/nio/NioSocketChannel.java

```diff
@@ -461,5 +470,21 @@ private NioSocketChannelConfig(NioSocketChannel channel, Socket javaSocket) {
         protected void autoReadCleared() {
             clearReadPending();
         }
+
+        @Override
+        public NioSocketChannelConfig setSendBufferSize(int sendBufferSize) {
+            super.setSendBufferSize(sendBufferSize);
+            // Multiply by 2 to give some extra space in case the OS can process write data faster than we can provide.
+            setMaxBytesPerGatheringWrite(getSendBufferSize() << 1);
```

Comment:
shouldn't we also call this in the constructor as we do for the native impls?

Reply:
sure